code/test-utils/src/test/java/com/adobe/marketing/tester/util/JSONAssertsTests.kt

```diff
@@ -0,0 +1,1648 @@
+
+        val actualJSONString = """
+        [[[]]]
+        """.trimIndent()
+        val expected = JSONArray(expectedJSONString)
+        val actual = JSONArray(actualJSONString)
+
+        assertEqual(expected, actual)
+        assertExactMatch(expected, actual)
+        assertTypeMatch(expected, actual)
+    }
+
+    @Test
+    fun `test special key name empty string`() {
+        val expectedJSONString = """
+        {
+            "": 1
+        }
+        """.trimIndent()
+
+        val actualJSONString = """
+        {
+            "": 1
+        }    
+        """.trimIndent()
+        val expected = JSONObject(expectedJSONString)
+        val actual = JSONObject(actualJSONString)
+
+        assertEqual(expected, actual)
+        assertExactMatch(expected, actual)
+        assertTypeMatch(expected, actual)
+    }
+
+    @Test
+    fun `test special key name backslash`() {
+        val expectedJSONString = """
+        {
+            "\\": 1
+        }
+        """.trimIndent()
+
+        val actualJSONString = """
+        {
+            "\\": 1
+        }    
+        """.trimIndent()
+        val expected = JSONObject(expectedJSONString)
+        val actual = JSONObject(actualJSONString)
+
+        assertEqual(expected, actual)
+        assertExactMatch(expected, actual)
+        assertTypeMatch(expected, actual)
+    }
+
+    @Test
+    fun `test special key name double backslash`() {
+        val expectedJSONString = """
+        {
+            "\\\\": 1
+        }
+        """.trimIndent()
+
+        val actualJSONString = """
+        {
+            "\\\\": 1
+        }    
+        """.trimIndent()
+        val expected = JSONObject(expectedJSONString)
+        val actual = JSONObject(actualJSONString)
+
+        assertEqual(expected, actual)
+        assertExactMatch(expected, actual)
+        assertTypeMatch(expected, actual)
+    }
+
+    @Test
+    fun `test special key name period`() {
+        val expectedJSONString = """
+        {
+            ".": 1
+        }
+        """.trimIndent()
+
+        val actualJSONString = """
+        {
+            ".": 1
+        }    
+        """.trimIndent()
+        val expected = JSONObject(expectedJSONString)
+        val actual = JSONObject(actualJSONString)
+
+        assertEqual(expected, actual)
+        assertExactMatch(expected, actual)
+        assertTypeMatch(expected, actual)
+    }
+
+    @Test
+    fun `test special key name inner period`() {
+        val expectedJSONString = """
+        {
+            "k.1.2.3": 1
+        }
+        """.trimIndent()
+
+        val actualJSONString = """
+        {
+            "k.1.2.3": 1
+        }    
+        """.trimIndent()
+        val expected = JSONObject(expectedJSONString)
+        val actual = JSONObject(actualJSONString)
+
+        assertEqual(expected, actual)
+        assertExactMatch(expected, actual)
+        assertTypeMatch(expected, actual)
+    }
+
+    @Test
+    fun `test special key name trailing period`() {
```

Comment:
since you are running same asserts for all these special characters I think you can write these as parameterized tests where you predefine the expected/actual values.
this will save some lines of code as well and show all the values tested in one place

Reply:
wow, after these changes 1k lines were gone! 🎉  
I left some comments on the ones that looked more intimidating, please review to see if they make sense to be kept in the expanded form.